src/refinement/ptq.jl

```diff
@@ -0,0 +1,61 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    PTQ()
+
+Refinement of triangle meshes.
+A triangle is subdivided into four triangles.
+"""
+struct PTQ <: RefinementMethod end
+
+function refine(mesh, ::PTQ)
+  primal_triangle_quadrisection(mesh, (p1, p2) -> Point((coordinates(p1) + coordinates(p2))/2))
+end
+
+function ordered_pair(i ,j)
+  i < j ? (i, j) : (j, i)
+end
+
+function primal_triangle_quadrisection(mesh, middle)
+  if !(eltype(mesh) <: Triangle)
+    throw(ArgumentError("The mesh is not triangle."))
```

Comment:
Can we generalize this refinement method to support arbitrary polygonal meshes?  I think we can call it EarRefinement, an algorihtm that clips ears of the poygon as triangles and then triangulates the interior polygon with FanTriangulation.

Reply:
By calling `triangulate()` at the beginning?

Reply:
That would be an interesting alternative as well.